primitives/src/currency.rs

```diff
@@ -0,0 +1,181 @@
+use crate::evm::EvmAddress;
+use bstringify::bstringify;
+use codec::{Decode, Encode};
+use sp_runtime::RuntimeDebug;
+use sp_std::{
+	convert::{Into, TryFrom, TryInto},
+	prelude::*,
+};
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+macro_rules! create_currency_id {
+    ($(#[$meta:meta])*
+	$vis:vis enum TokenSymbol {
+        $($(#[$vmeta:meta])* $vname:ident = $val:literal,)*
+    }) => {
+        $(#[$meta])*
+        $vis enum TokenSymbol {
+            $($(#[$vmeta])* $vname = $val,)*
+        }
+
+        impl TryFrom<u8> for TokenSymbol {
+            type Error = ();
+
+            fn try_from(v: u8) -> Result<Self, Self::Error> {
+                match v {
+                    $($val => Ok(TokenSymbol::$vname),)*
+                    _ => Err(()),
+                }
+            }
+        }
+
+		impl TryFrom<Vec<u8>> for CurrencyId {
+			type Error = ();
+			fn try_from(v: Vec<u8>) -> Result<CurrencyId, ()> {
+				match v.as_slice() {
+					$(bstringify!(vname) => Ok(CurrencyId::Token(TokenSymbol::$vname)),)*
+					_ => Err(()),
+				}
+			}
+		}
+
+		#[test]
+		fn generate_token_resources() {
+			#[allow(non_snake_case)]
+			#[derive(Serialize, Deserialize)]
+			struct Token {
+				name: String,
+				symbol: String,
+				currencyId: String,
+			}
+
+			let tokens = vec![
+				$(
+					Token {
+						name: stringify!($vname).to_string(),
+						symbol: stringify!($vname).to_string(),
+						currencyId: $val.to_string(),
+					},
+				)*
+			];
+			frame_support::assert_ok!(std::fs::write("tokens.json", serde_json::to_string_pretty(&tokens).unwrap()));
```

Comment:
can make path to the predeploy-contracts submodule directly so we don't need to move files. we can just run this and commit the change

Reply:
Add `scripts/generate-tokens-and-predeploy-contracts.sh` to do it.